addons/sale/migrations/9.0.1.0/post-migration.py

```diff
@@ -55,12 +55,139 @@ def set_crm_team_message_types(env):
+                price_subtotal = %s
+            WHERE id = %s""", (
+                taxes['total_included'] - taxes['total_excluded'],
+                taxes['total_included'],
+                taxes['total_excluded'],
+                line.id,
+            )
+        )
+
+
+def set_so_line_computed_rest(env):
+    """Emulate the computation of the rest of the computed fields through
+    equivalente SQL querys.
+    """
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE sale_order_line
+        SET price_reduce = CASE
+            WHEN abs(product_uom_qty) > 0.00001
+                THEN price_subtotal / product_uom_qty
+            ELSE 0.0
+        END""")
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE sale_order_line sol
+        SET qty_invoiced = sub.qty
+        FROM (
+            SELECT rel.order_line_id,
+                SUM(CASE
+                    WHEN ai.type = 'out_invoice'
+                        THEN ail.quantity / u.factor * u2.factor
+                    WHEN ai.type = 'out_refund'
+                        THEN -ail.quantity / u.factor * u2.factor
+                    ELSE 0.0
+                END) AS qty
+            FROM sale_order_line_invoice_rel rel
+                JOIN sale_order_line sol2 ON
+                    rel.order_line_id = sol2.id
+                JOIN account_invoice_line ail ON
+                    rel.invoice_line_id = ail.id
+                JOIN account_invoice ai ON ai.id = ail.invoice_id
+                JOIN product_uom u on (u.id = ail.uom_id)
+                JOIN product_uom u2 on (u2.id = sol2.product_uom)
+            WHERE ai.state != ('cancel')
+            GROUP BY rel.order_line_id
+        ) AS sub
+        WHERE sol.id = sub.order_line_id""")
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE sale_order_line sol
+        SET qty_to_invoice = (CASE
+            WHEN so.state IN ('sale', 'done')
+                THEN (CASE
+                    WHEN pt.invoice_policy = 'order'
+                        THEN sol.product_uom_qty - sol.qty_invoiced
+                    ELSE sol.qty_delivered - sol.qty_invoiced
+                END)
+            ELSE 0.0
+        END)
+        FROM sale_order_line sol2
+            JOIN sale_order so ON so.id = sol2.order_id
+            JOIN product_product pp ON pp.id = sol2.product_id
+            JOIN product_template pt ON pt.id = pp.product_tmpl_id
+        WHERE sol2.id = sol.id""")
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE sale_order_line sol
+        SET invoice_status = CASE
+            WHEN sol.state IN ('sale', 'done') THEN 'no'
+            WHEN abs(sol.qty_to_invoice) > 0.00001 THEN 'to invoice'
+            WHEN sol.state = 'sale' AND pt.invoice_policy = 'order' AND
+                (sol.qty_delivered - sol.product_uom_qty) > 0.00001
+                    THEN 'upselling'
+            WHEN sol.qty_invoiced >= sol.product_uom_qty THEN 'invoiced'
```

Comment:
This one has a little bug also compared to: https://github.com/OCA/OpenUpgrade/blob/bd1a1d7b04b1d5ea5d94b931443d61fc71d810a2/addons/sale/sale.py#L500-L501

Change to fix it:

```suggestion
            WHEN ROUND(sol.qty_invoiced, %(uom_precision)s) = ROUND(sol.product_uom_qty, %(uom_precision)s) THEN 'invoiced'
```